### PR TITLE
Enable backups and encrypt username and password

### LINKED ISSUE #1508: Enable backups
[Backups have been disable](https://github.com/openhab/openhab-android/blob/master/mobile/src/main/AndroidManifest.xml#L24) to make sure that usernames and passwords aren't saved in clear text to Google Drive or any other backup service.
Usernames and passwords should be moved to a second shared prefs file, that will be encrypted (https://developer.android.com/reference/androidx/security/crypto/EncryptedSharedPreferences) and excluded from backups.

### PR DESCRIPTION
**Pull request by mueller-ma**
Closes #1508

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt [hunk 1]
```diff
@@ -57,16 +70,24 @@ class CustomInputTypePreference constructor(context: Context, attrs: AttributeSe
                 }
             }
             super.onBindDialogView(view)
+            if (arguments?.getBoolean(KEY_IS_PREF_SECRET) == true) {
+                editor?.setText(context?.getSecretPrefs()?.getString(this.preference.key))
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt]: Where/how is the value written to the secret prefs?

I think a better approach would be writing a (trivial) `PreferenceDataStore` around the secret prefs and attaching that to the respective prefs from within `PreferenceActivity`.

Comment (maniac103) [reply mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt]: Implementation to above suggestion: https://github.com/maniac103/openhab.android/commits/backup ... IMHO it simplifies logic quite a bit.

Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/preference/CustomInputTypePreference.kt]: Indeed, that's better.

**Comment by mueller-ma** [comment]
I'd add some hint to the user that he has to re-enter the credentials after restoring. Is a simple toast enough?

**Comment by maniac103** [comment]
Maybe a Snackbar with button to open settings?

**Comment by mueller-ma** [comment]
I'd rather go for a notification since the snackbars on startup are easily dismissed by other snackbars, e.g. those for showing if connecting to local or remote server.

**Pull request merged**